_posts/2016-09-16-angular-2-animation-important-concepts

```diff
@@ -0,0 +1,236 @@
+---
+layout:     post
+title:      "Angular 2 Animations - Foundation Concepts"
+
+date: 2016-06-08
+imageUrl: '/images/banner/angular-2-component-animations.jpg'
+
+summary: "Animation in Angular 2 is now easy and more intuitive... Learn foundational animation concepts and start animating your Angular 2 components!"
+
+categories:
+  - angular
+
+tags:
+  - angular2
+  - animation
+  - components
+  - relative paths
+
+topic: components
+
+author: thomas_burleson
+---
+
+
+### Our Scenario
+
+Here is a simple Angular 2 component with hide and show features... but no animations (yet):
+
+{% highlight js %}
+{% raw %}
+@Component({
+  selector: 'my-fader',
+    template: `
+    <div *ngIf="visibility == 'shown'" >
+      <ng-content></ng-content>
+      Can you see me? 
+    </div>
+  `
+})
+export class MyComponent implements OnChanges {
+  visibility = 'shown';
+
+  @Input() isVisible : boolean = true;
+
+  ngOnChanges() {
+   this.visibility = this.isVisible ? 'shown' : 'hidden';
+  }
+}
+{% endraw %}
+{% endhighlight %}
+
+This component simply publishes an **@Input** `isVisible` property; which allows other external components to show/hide the text content... without any animations.
+
+<iframe style="width: 100%; height: 600px" src="http://embed.plnkr.co/chNcuU7IoIrN22DF" frameborder="0" allowfullscren="allowfullscren"></iframe>
+
+
+### Configure Component Animations
+
+We want the `my-fader` component to **fade-in** or **fade-out** its text content. And we want to *animate* those fades effects.
+
+> The essential take-away Animation concept is that **Angular 2 Animations** are triggered on component state changes. 
+Developers should consider state changes simply as value changes in a property of the component instance.
+
+To start animating, let's first add animation metadata to our component.
+
+Since our component has a `visibility` property whose state can change between `shown` and `hidden`, 
+let’s configure animations to trigger and animate during each value change.
+
+{% highlight js %}
+{% raw %}
+animations: [
+  trigger(’visibilityChanged', [
+  state('shown' , style({ opacity: 1 })), 
+  state('hidden', style({ opacity: 0 }))
+  ])
+]
+{% endraw %}
+{% endhighlight %}
+
+But what does the above mean... and why is this syntax used?
+
+The techno-speak above is saying that when the `visibilityChanged` property changes to the value == 'shown', 
```

Comment:
Hah.. just realised that's gonna be clarified later. So how about we add something like:

"Now, we might wonder where `visibilityChanged` comes from, because our property is called just `visibility`. Hold your horses, we'll clarify that in a second"

Doesn't have to be this wording, but just a quick heads-up for the reader that we are aware of the fact that `visibilityChanged` is not the same as `visibility`